File: codigo/elixir/aula2.ex
defmodule Aula2 do
  # definicao recursiva de soma e multiplicacao
  def soma(n, 0), do: n
  def soma(n, m), do: soma(n+1, m-1)

  def mult(_n, 0), do: 0
  def mult( n, m), do: n + mult(n, m-1)

  # alguns exemplos de listas
  def listas() do
    lista1 = [1 | []]
    lista2 = [2 | lista1]
    lista1 == [1] and lista2 == [2, 1]
  end

  # tamanho de listas
  def tamanho([]), do: 0
  def tamanho([_ | t]), do: 1 + tamanho(t)
end

File: codigo/erlang/aula2.erl
-module(aula2).

-export([soma/2, mult/2, tamanho/1]).

% funcoes recursivas para soma e multiplicacao
soma(M, 0) -> M;
soma(M, N) -> soma(M+1, N-1).

mult(_M, 0) -> 0;
mult( M, N) -> M + mult(M, N-1).

% alguns exemplos de listas
listas() ->
    Lista1 = [ 1 | [] ],
    Lista2 = [ 2 | Lista1 ],
    ([1] == Lista1) and ([2, 1] == Lista2).

% tamanho de listas
tamanho([]) -> 0;
tamanho([_ | T]) -> 1 + tamanho(T).
